test/relay_hub_test.js

```diff
@@ -414,13 +414,79 @@ contract("RelayHub", function (accounts) {
         assert.equal(address, res.logs[0].args.relay.toLowerCase())
         assert.equal(snitching_account, res.logs[0].args.sender)
         increaseTime(dayInSec)
+        await claim_snitch_reward(address, snitching_account_initial_balance, stake, res)
+    });
+    /**
+     * 
+     * @param {*} address - relay that will be penalized
+     * @param {*} snitching_account_initial_balance 
+     * @param {*} stake - relay's initial stake
+     * @param {*} res - receipt on the 'penalize' request
+     */
+    let claim_snitch_reward = async function(address, snitching_account_initial_balance, stake, res) {
         let res2 = await rhub.unstake(address,{from:snitching_account, gasPrice: gasPricePenalize,gasLimit: gas_limit_any_value})
 
         let balance_of_acc7 = await web3.eth.getBalance(snitching_account);
         let expected_balance_after_penalize = new Big(snitching_account_initial_balance).add(stake[0]).sub(res.receipt.gasUsed * gasPricePenalize).sub(res2.receipt.gasUsed * gasPricePenalize)
         assert.equal(expected_balance_after_penalize, balance_of_acc7);
+    }
+
+    let asyncForEach = async function (array, callback) {
+        for (let index = 0; index < array.length; index++) {
+          await callback(array[index], index, array);
+        }
+    }
+
+    it("should penalize relay for calling any non-RelayHub address or a method not whitelisted inside hub", async function () {
+        // A call to a method that is not whitelisted for the relay to use
+        let data1 = rhub.contract.methods.remove_relay_by_owner(testutils.zeroAddr).encodeABI()
+        let data2 = sr.contract.methods.emitMessage("Hello SampleRecipient!").encodeABI()
+        let arr = [data1, data2]
+        await asyncForEach(arr, async function (data) {
+            await register_new_relay_with_privkey(rhub, one_ether, dayInSec, 120, "hello", accounts[0], web3, privKey);
+            let address = "0x" + ethUtils.privateToAddress(privKey).toString('hex')
+            let stake = await rhub.relays(address);
+            assert.equal(one_ether, stake[0]);
+
+            let illegalTransaction = new ethJsTx({
```

Comment:
why another illegal transaction inside the loop?

Reply:
An outside variable is of "string" type and is a tx data (not called 'illegalTransaction')
An inside variable is of class 'EthereumTx' so I can create a signature for it.